src/main/java/com/amihaiemil/docker/Containers.java

```diff
@@ -75,10 +76,25 @@ Container create(
     Container create(final JsonObject container) throws IOException;
     
     /**
-     * Return all the Containers, not only the running ones.
+     * Return all Containers, not only the running ones.
      * @return Iterator over all the containers.
      */
     Iterator<Container> all();
+
+    /**
+     * Whether to request the size of containers (fields SizeRw and SizeRootFs).
+     * @param withSize Return the size of containers (SizeRw and SizeRootFs).
+     * @return Containers with modified size flag.
+     */
+    Containers withSize(final boolean withSize);
+
+    /**
+     * Filter these Containers.
+     * @param filters Filters to apply.
+     * @return Filtered containers.
+     * @see <a href="https://docs.docker.com/engine/api/v1.35/#operation/ContainerList">Docker API Docs</a>
+     */
+    Containers filter(Map<String, Iterable<String>> filters);
```

Comment:
@milux I think that this should be implemented as a decorator called `Filtered`

Reply:
Sorry, but I'm a bit puzzled because I did my implementation in line with the patterns I found in `RtImages` and `ListedImages`, and now you suggest a completely different pattern that has not been used in the library AFAICT.
Also, decorator patterns are applicable to one class only, because a decorator cannot expose functionality of multiple classes, right?
However, I think we could indeed improve code quality by defining this method in a separate interface (e.g. `Filterable`) and implementing that in all classes with that functionality.

Reply:
OK, I think we have to get some things straight here, because afaik the decorator pattern is an approach that simply doesn't make any sense here. It's almost like saying "let's use the observer pattern here". Please let me explain why I think so:
We could use the decorator pattern, whereas our decorator contains more or less precisely the same code as `ListedContainers`, plus a bunch of proxy methods that forward calls to the original `RtContainer` instance. We **cannot remove `all()` and `iterator()`**, because they need information about the filters and/or size flag to construct the API request, and the underlying implementation is per definition unaware of the wrapping decorator when using this pattern. Thus, we end up with a lot of **code duplication**, because we need to implement `all()` and `iterator()` twice, once for `RtContainer` and once for the decorator.
Maybe I'm an idiot and got your excellent idea all wrong, but I believe there might be an implicit error in your imagination about how this should work. If you still think decorators are a good idea, then please elaborate a bit more on this.